File: frontend/lib/features/public_messages/data/models/requests/public_message.dart
class PublicMessageUpdateRequestModel {
  final String content;

  const PublicMessageUpdateRequestModel({
    required this.content,
  });

  Map<String, dynamic> toJson() {
    return {
      'content': content,
    };
  }
}

class PublicMessageCreateRequestModel {
  final String? habitId;
  final String? challengeId;
  final String? repliesTo;
  final String? threadId;
  final String content;

  const PublicMessageCreateRequestModel({
    required this.habitId,
    required this.challengeId,
    required this.repliesTo,
    required this.threadId,
    required this.content,
  });

  Map<String, dynamic> toJson() {
    return {
      'habit_id': habitId,
      'challenge_id': challengeId,
      'replies_to': repliesTo,
      'thread_id': threadId,
      'content': content,
    };
  }
}

File: backend/api/src/features/public_discussions/application/dto/requests/public_message.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Deserialize)]
pub struct GetPublicMessageParams {
    pub message_id: Uuid,
}

#[derive(Deserialize)]
pub struct UpdatePublicMessageParams {
    pub message_id: Uuid,
}

#[derive(Deserialize)]
pub struct DeletePublicMessageParams {
    pub message_id: Uuid,
    pub deleted_by_admin: bool,
}

#[derive(Deserialize)]
pub struct GetPublicMessagesParams {
    pub habit_id: Option<Uuid>,
    pub challenge_id: Option<Uuid>,
}

#[derive(Deserialize)]
pub struct GetPublicMessageRepliesParams {
    pub message_id: Uuid,
}

#[derive(Deserialize)]
pub struct GetPublicMessageParentsParams {
    pub message_id: Uuid,
}

#[derive(Deserialize, Serialize)]
pub struct PublicMessageUpdateRequest {
    pub content: String,
}

#[derive(Deserialize, Serialize)]
pub struct PublicMessageCreateRequest {
    pub habit_id: Option<Uuid>,
    pub challenge_id: Option<Uuid>,
    pub replies_to: Option<Uuid>,
    pub content: String,
    pub thread_id: Option<Uuid>,
}
